syncmaster/worker/handlers/file/samba.py

```diff
@@ -0,0 +1,84 @@
+# SPDX-FileCopyrightText: 2023-2024 MTS PJSC
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import os
+from typing import TYPE_CHECKING
+
+from onetl.connection import Samba, SparkLocalFS
+from onetl.file import FileDFReader, FileDFWriter, FileDownloader, FileUploader
+
+from syncmaster.dto.connections import SambaConnectionDTO
+from syncmaster.worker.handlers.file.base import FileHandler
+
+if TYPE_CHECKING:
+    from pyspark.sql import DataFrame, SparkSession
+
+
+class SambaHandler(FileHandler):
+    connection_dto: SambaConnectionDTO
+
+    def connect(self, spark: SparkSession) -> None:
+        self.connection = Samba(
+            host=self.connection_dto.host,
+            port=self.connection_dto.port,
+            share=self.connection_dto.share,
+            protocol=self.connection_dto.protocol,
+            domain=self.connection_dto.domain,
+            user=self.connection_dto.user,
+            password=self.connection_dto.password,
+            auth_type=self.connection_dto.auth_type,
+        ).check()
+        self.local_connection = SparkLocalFS(
+            spark=spark,
+        ).check()
+
+    def read(self) -> DataFrame:
```

Comment:
`read` and `write` implementations looks the same for SFTP, FTP, FTPS, Samba. Could we more this to a new base class.